### PR TITLE
feat(schema): do schema versioning and make backup non-blocking for indexing

### PR DESCRIPTION
**Pull request by NamanJain8**
**Issue:**
We used to write schema/types at `timestamp=1`. That single bit violates the Snapshot Isolation. Hence, we couldn't run schema updates and other operations like backup concurrently. This is an issue in a multi-tenant cluster because if the backup is running we couldn't update the schema. 
**Fix:**
This PR writes the schema/types at the `startTs` of the transaction and enables the schema updates to be made even if the backup is running.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7852)

### CODE REVIEW

File: posting/index.go [hunk 1]
```diff
@@ -561,7 +561,6 @@ func (r *rebuilder) Run(ctx context.Context) error {
 		WithNumVersionsToKeep(math.MaxInt32).
 		WithLogger(&x.ToGlog{}).
 		WithCompression(options.None).
-		WithEncryptionKey(x.WorkerConfig.EncryptionKey).
```
Comment (jarifibrahim) [inline posting/index.go:564]: Why do we need to remove this?

File: systest/backup/filesystem/backup_test.go [hunk 2]
```diff
@@ -113,7 +112,6 @@ func TestBackupOfOldRestore(t *testing.T) {
 
 	// Clean the cluster and try restoring the backups created above.
 	testutil.DropAll(t, dg)
-	time.Sleep(2 * time.Second)
```
Comment (jarifibrahim) [inline systest/backup/filesystem/backup_test.go:116]: Don't we need these anymore? The DropAll could take a few minutes to be effective.

File: worker/draft.go [hunk 3]
```diff
@@ -606,7 +623,7 @@ func (n *node) applyMutations(ctx context.Context, proposal *pb.Proposal) (rerr
 		}
 
 		for _, tupdate := range proposal.Mutations.Types {
-			if err := runTypeMutation(ctx, tupdate); err != nil {
+			if err := runTypeMutation(ctx, tupdate, startTs); err != nil {
```
Comment (jarifibrahim) [inline worker/draft.go:626]: The startTs is actually the `MaxAssigned`. See line 605. You might end up writing data at the wrong timestamp.

**Review by jarifibrahim — COMMENTED** [review]
Got some comments.

File: posting/index.go (hunk 1 above)
Comment (NamanJain8) [reply posting/index.go:564]: We are already setting the encryption key a few lines below. This was redundant.

File: systest/backup/filesystem/backup_test.go (hunk 2 above)
Comment (NamanJain8) [reply systest/backup/filesystem/backup_test.go:116]: I think this should not be needed. Why would DropAll take few minutes to be effective?

**Review by manishrjain — APPROVED** [review]
:lgtm: Do test for predicate moves and snapshots, etc.

> Reviewed 13 of 13 files at r1.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7852#-:-MaYz1QlCiH5b5WjnV3P:bj9eq8e)* status: all files reviewed, 2 unresolved discussions (waiting on @NamanJain8 and @vvbalaji-dgraph)

___
*[worker/draft.go, line 230 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/7852#-MaYy8fg4CCDuHjhyPwp:-MaYy8fg4CCDuHjhyPwq:b-bx0fi3) ([raw file](https://github.com/dgraph-io/dgraph/blob/f4122592e432a4922dc40f63d8d2d7ff1b0d7462/worker/draft.go#L230)):*
> ```Go
> 	case opIndexing:
> 		for otherId, otherCloser := range n.ops {
> 			if otherId == opBackup {
> ```

Ideally, we should look at the backup timestamp. If indexing has a higher ts, then it's OK. If not, then not.

File: systest/backup/filesystem/backup_test.go (hunk 2 above)
Comment (NamanJain8) [reply systest/backup/filesystem/backup_test.go:116]: I will remove these sleep for now. If they are needed, or tests become flaky, then we can bring them back.

**Review by NamanJain8 — COMMENTED** [review]
> Dismissed @jarifibrahim and @manishrjain from 2 discussions.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7852)* status: 10 of 13 files reviewed, all discussions resolved (waiting on @manishrjain and @vvbalaji-dgraph)

___
*[worker/draft.go, line 230 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/7852#-MaYy8fg4CCDuHjhyPwp:-MabWHKl0z0mVaocFNMr:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/f4122592e432a4922dc40f63d8d2d7ff1b0d7462/worker/draft.go#L230)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

Ideally, we should look at the backup timestamp. If indexing has a higher ts, then it's OK. If not, then not.
</blockquote></details>

Done.
___
*[worker/draft.go, line 626 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/7852#-MaXr_c-AvzY9Ly2FV1r-r1-626:-MabWKY37eIIhpRCAcVc:b-ybdfpz) ([raw file](https://github.com/dgraph-io/dgraph/blob/f4122592e432a4922dc40f63d8d2d7ff1b0d7462/worker/draft.go#L626)):*
<details><summary><i>Previously, jarifibrahim (Ibrahim Jarif) wrote…</i></summary><blockquote>

The startTs is actually the `MaxAssigned`. See line 605. You might end up writing data at the wrong timestamp.
</blockquote></details>

We should use the same timestamp as the one used for index rebuild (`MaxAssigned`). Max assigned is used here to index all the data that has been committed up until this point.

**Pull request merged**